keps/sig-scheduling/2458-node-resource-score-strategy/README.md

```diff
@@ -0,0 +1,674 @@
+when running a mix of serving and batch workloads on the same cluster. We believe that the
+changes we make here will enable this long term vision.
+
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+- Allow users to configure resource fit preferences using a single plugin configuration
+- Deprecate resource-based scoring plugins that implement individual strategies, specifically: 
+  - `NodeResourcesLeastAllocated`
+  - `NodeResourcesMostAllocated`
+  - `RequestedToCapacityRatio`
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+- Allow users to express resource fit preferences via the pod spec. This is left as a
+  follow up work that will be done under a different KEP.  
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+-->
+
+#### Story 1
+As a cluster operator, I want an easy way to configure the scoring behavior of the scheduler
+with respect to node resources. 
+
+<!--
+
+### Notes/Constraints/Caveats (Optional)
+
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+Potentially reduce the number of allowed combinations too much. This can be mitigated
+by adding more configuration strategies wherever applicable. 
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+Define a new `ScoringStrategy` type as follows:
```

Comment:
Would this be better as `ResourceScoringStrategy`? Or is this planned to be used for other scoring plugins besides resource allocation?

Reply:
may be, lets leave that to the api review.